cmd/keymasterd/storage.go

```diff
@@ -20,6 +20,26 @@ const userProfileSuffix = ".gob"
 const profileDBFilename = "userProfiles.sqlite3"
 const cachedDBFilename = "cachedDB.sqlite3"
 
+func (state *RuntimeState) expandStorageUrl() error {
+	config := &state.Config.ProfileStorage
+	if config.AwsSecretId == "" {
+		return nil
+	}
+	metadataClient, err := getMetadataClient()
+	if err != nil {
+		return err
+	}
+	secrets, err := getAwsSecret(metadataClient, config.AwsSecretId)
+	if err != nil {
+		return err
+	}
+	config.StorageUrl = os.Expand(config.StorageUrl,
```

Comment:
While this works I am not very happy on leaving the secrets on the environment. Any other way? Also you give users too much leeeway.

Reply:
The secrets are not in the environment; they are only in memory. The `os.Expand()` function is just a variable expander.

Reply:
The approach I took gives users the flexibility to store the hostname of the Postgres server in AWS Secrets Manager too, if they wish, which I think is a valuable feature. Note that `os.Expand()` _does not_ pull from the environment.

Reply:
I actually misread the docs.. I was reading at expandEnv.. not expand.